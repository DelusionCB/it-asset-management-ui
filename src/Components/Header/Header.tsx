import React, {useState} from 'react';
import './index.scss'
import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavItem, Button,
} from 'reactstrap';
import {Link, useNavigate} from 'react-router-dom'
import {redirectToAction} from '../../Utils/routing';
import {useTranslation} from 'react-i18next';

function Header (args: JSX.IntrinsicAttributes & JSX.IntrinsicClassAttributes<Navbar>): JSX.Element {
    const [isOpen, setIsOpen] = useState(true);
    const navigate = useNavigate()
    const {t} = useTranslation()

    function toggleOpen (): void {
        setIsOpen(!isOpen);
    }

    function redirect (action: string = 'create', form: boolean = false): void {
        if (form) {
            redirectToAction(navigate, action);
        } else {
            navigate(action)
        }
    }

    return (
        <header className='main-navbar'>
            <div className='bar'>
                <Link
                    className='bar__logo'
                    to='/'
                    relative="path"
                    aria-label='Linkki etusivulle, Varhan logo'
                />
                <div className='bar__login'>
                    <Button role='link' disabled={true}>
                        <i className="bi bi-person-fill-lock" />
                        <span>{t('user.login')}</span>
                    </Button>
                </div>
            </div>

            <Navbar role='navigation' className='mappi-bar' expand='xl'>
                <NavbarToggler onClick={() => { toggleOpen() }} />
                <Collapse isOpen={!isOpen} navbar>
                    <ul className='mappi-bar__links'>
                        <NavItem>
                            <Button
                                className='navlink-button'
                                role='link'
                                aria-label={`${t('create-new')}`}
                                onClick={() => { redirect('create', true) }}
                            >
                                <span>{t('create-new')}</span>
                                <i className="bi bi-plus" />
                            </Button>
                        </NavItem>
                        <NavItem>
                            <Button
                                className='navlink-button'
                                role='link'
                                aria-label={`${t('archive-fetch')}`}
                                onClick={() => { redirect('search', false) }}
                            >
                                <span>{t('archive-fetch')}</span>
                                <i className='bi bi-search' />
                            </Button>
                        </NavItem>
                        <NavItem>
                            <Button
                                className='navlink-button'
                                role='link'
                                disabled={true}
                                aria-label={`${t('archive-fetch')}`}
                                onClick={() => { redirect('search', false) }}
                            >
                                <span>Tilastot</span>
                                <i className='bi bi-graph-up-arrow' />
                            </Button>
                        </NavItem>
                        <NavItem>
                            <Button
                                className='navlink-button'
                                role='link'
                                disabled={true}
                                aria-label={`${t('archive-fetch')}`}
                                onClick={() => { redirect('search', false) }}
                            >
                                <span>Hallintapaneeli</span>
                                <i className='bi bi-wrench' />
                            </Button>
                        </NavItem>
                    </ul>
                </Collapse>
            </Navbar>
        </header>
    );
}

export default Header;
