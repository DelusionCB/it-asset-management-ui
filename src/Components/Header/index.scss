.main-navbar {
    .container-fluid {
        flex-direction: row-reverse;
    }
    button {
        font-weight: bold;
        font-size: 1.25rem;
        background-color: transparent !important;
        border: none;
        &:hover, &:focus, &:active {
            color: inherit;
            text-decoration: underline var(--secondary-color);
        }
    }
    .bar {
        width: 100%;
        background: var(--primary-color);
        padding: 12px 24px 12px 24px;
        display: inline-flex;
        justify-content: space-between;
        &__login {
            display: inline-flex;
            align-items: center;
            button {
                display: flex;
                align-items: center;
                text-decoration: none;
                &:hover, &:focus {
                    text-decoration: underline;
                    color: var(--white-color);
                }
            }
        }
    }

    .mappi-bar {
        height: 64px;
        width: 100%;
        background: var(--white-color);
        z-index: 1;
        border-bottom: rgba($color: #404040, $alpha: 0.75) 2px solid !important;
        &__links {
            display: contents;
            color: var(--black-color);
            .nav-item {
                margin-right: 1rem;
                display: block;
                .navlink-button {
                    text-decoration: none;
                    font-weight: bold;
                    padding-top: 12px;
                    padding-bottom: 12px;
                    position: relative;
                    color: var(--black-color);
                    font-size: 1.25rem;
                    i {
                        margin-left: 5px;
                    }
                    &.active, &:hover, &:focus {
                        background-color: rgba(0, 0, 0, 0.08);
                        outline: solid 2px var(--black-color);
                    }
                }
            }
        }
    }
    .bar__logo {
        background-image: url('../../Assets/Images/varha_logo_white.webp');
        height: 90px;
        background-size: contain;
        background-repeat: no-repeat;
        cursor: pointer;
        width: 100%;
    }
}

// Hamburger for Mobile
.navbar-toggler-icon {
    background-image: url("data:image/svg+xml;charset=utf8,%3Csvg viewBox='0 0 32 32' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath stroke='rgb(0,0,0)' stroke-width='2' stroke-linecap='round' stroke-miterlimit='10' d='M4 8h24M4 16h24M4 24h24'/%3E%3C/svg%3E") !important;
}

//Mobile
@media only screen and (max-width: 1199px) {
    .main-navbar {
        height: auto;
        .bar {
            height: auto;
            flex-wrap: nowrap;
            &__login {
                .btn {
                    white-space: normal;
                    display: flex;
                    align-items: center;
                    height: auto;
                }
            }
        }
        .mappi-bar {
            height: auto;
            width: auto;
            &__links {
                display: block;
                text-align: center;
                padding: 1rem;
                .nav-item {
                    margin-right: 0;
                    margin-top: 1rem;
                    &:first-child {
                        margin-top: 0;
                    }
                }
            }
        }
    }
}
