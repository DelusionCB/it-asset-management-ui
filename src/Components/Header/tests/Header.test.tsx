import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import Header from '../Header';
import {BrowserRouter} from 'react-router-dom'
import '@testing-library/jest-dom'

describe('<Header />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <Header />
            </BrowserRouter>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
