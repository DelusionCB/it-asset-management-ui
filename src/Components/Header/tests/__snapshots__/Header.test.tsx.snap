// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Snapshot renders correctly 1`] = `
<header
  className="main-navbar"
>
  <div
    className="bar"
  >
    <a
      aria-label="Linkki etusivulle, Varhan logo"
      className="bar__logo"
      href="/"
      onClick={[Function]}
    />
    <div
      className="bar__login"
    >
      <button
        className="btn btn-secondary disabled"
        disabled={true}
        onClick={[Function]}
        role="link"
      >
        <i
          className="bi bi-person-fill-lock"
        />
        <span>
          user.login
        </span>
      </button>
    </div>
  </div>
  <nav
    className="mappi-bar navbar navbar-expand-xl"
    role="navigation"
  >
    <div
      className="container-fluid"
    >
      <button
        aria-label="Toggle navigation"
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={{}}
      >
        <ul
          className="mappi-bar__links"
        >
          <li
            className="nav-item"
          >
            <button
              aria-label="create-new"
              className="navlink-button btn btn-secondary"
              onClick={[Function]}
              role="link"
              type="button"
            >
              <span>
                create-new
              </span>
              <i
                className="bi bi-plus"
              />
            </button>
          </li>
          <li
            className="nav-item"
          >
            <button
              aria-label="archive-fetch"
              className="navlink-button btn btn-secondary"
              onClick={[Function]}
              role="link"
              type="button"
            >
              <span>
                archive-fetch
              </span>
              <i
                className="bi bi-search"
              />
            </button>
          </li>
          <li
            className="nav-item"
          >
            <button
              aria-label="archive-fetch"
              className="navlink-button btn btn-secondary disabled"
              disabled={true}
              onClick={[Function]}
              role="link"
              type="button"
            >
              <span>
                Tilastot
              </span>
              <i
                className="bi bi-graph-up-arrow"
              />
            </button>
          </li>
          <li
            className="nav-item"
          >
            <button
              aria-label="archive-fetch"
              className="navlink-button btn btn-secondary disabled"
              disabled={true}
              onClick={[Function]}
              role="link"
              type="button"
            >
              <span>
                Hallintapaneeli
              </span>
              <i
                className="bi bi-wrench"
              />
            </button>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</header>
`;
